src/sbvr-api/control-flow.ts

```diff
@@ -0,0 +1,60 @@
+import * as _ from 'lodash'
+import * as Promise from 'bluebird'
+
+// TODO: We force the return type here to merge `Promise<U> | Promise<U[]>` into `Promise<U | U[]>,
+// which is equivalent for our uses but fixes issues where typing were being rejected
+export const liftP = <T, U>(fn: (v: T) => U | Promise<U>): (a: T | T[]) => Promise<U | U[]> => {
```

Comment:
I agree we should be marking this as a proper function, I know we try to minimise the difference between the CS and TS versions when converting, but let's not make things more complicated just to keep meeting that end.

Reply:
Why would it make it simpler to use `function...` instead? It would just mean that `this` could be anything, not that it matters either way when we don't use it

Reply:
It's simpler because top level functions don't have "this" anyway, so it doesn't matter that they're functions, and you get better debug-ability for free due to the function name being visible in stack traces.

Reply:
The function name will be visible with the arrow style as well.  And I would expect the top level function to have a `this` variable, depending on the context it's called in of course, eg
```
import * as controlFlow from 'control-flow'
controlFlow.liftP(...)
``
I would expect to have `this` be `controlFlow` for the `function` style (certainly if using an ES5 target, maybe true ES6 modules have a special case for this)

But my real problem is that I really don't get the argument that it's simpler, it's an alternative for this function for sure as it doesn't change anything, but it doesn't seem in any way simpler to use `function liftP (...) {` vs `liftP = (...) => {` - so I'm against changing it for the reasoning that it's "simpler" because I don't see the argument or benefit from doing so and I'd like to understand why you're saying it is